src/_pytest/config/__init__.py

```diff
@@ -1094,13 +1094,26 @@ def _validate_plugins(self) -> None:
         if not required_plugins:
             return
 
+        # Imported lazily to improve start-up time.
+        from packaging.version import Version
+        from packaging.requirements import InvalidRequirement, Requirement
+
         plugin_info = self.pluginmanager.list_plugin_distinfo()
-        plugin_dist_names = [dist.project_name for _, dist in plugin_info]
+        plugin_dist_info = {dist.project_name: dist.version for _, dist in plugin_info}
 
         missing_plugins = []
-        for plugin in required_plugins:
-            if plugin not in plugin_dist_names:
-                missing_plugins.append(plugin)
+        for required_plugin in required_plugins:
```

Comment:
As a follow-up we might want to consider caching this depending on the import time

Reply:
It looks like we do end up caching it.
We get required plugins from here https://github.com/pytest-dev/pytest/blob/master/src/_pytest/config/__init__.py#L1105
That method ends up here https://github.com/pytest-dev/pytest/blob/master/src/_pytest/config/__init__.py#L1171 which looks like it maintains a cache for ini keys/values

Or are you referring to the `import` statements ( in which case I don't know if we cache this or not ). This file doesn't seem to

Reply:
The cache would simply be a way to avoid doing the version check and import often, if the expense seems problematic

Reply:
I'll make a follow up issue later today. I think we should apply this import cache everywhere we do conditional imports

Reply:
I think you misunderstood, we can't cache imports

We can however cache certain outcomes to make initial import unnecessary in case of high import cost

Reply:
I see. Just to clarify are you suggesting we cache the name of imports we’ve made ( so we don’t import multiple times ) or are you suggesting we cache the list of required plugins

Reply:
A no to both, its not clear what to put into such a cache and how to structure it,

The result we want to get is that for the current environment with the current installed packages the declared requirements are met, it seems a good amount of tinkering will be needed to make it reasonable

Reply:
Thanks @RonnyPfannschmidt I understand and agree on all points. Let’s move the conversation to the follow up issue ( when I raise it )

Reply:
resolving as per https://github.com/pytest-dev/pytest/issues/7369